.github/workflows/build.yml

```diff
@@ -8,27 +8,27 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8, 3.9, "3.10", "3.11"]
+        python-version: [3.9, "3.10", "3.11"]
+      fail-fast: false
     steps:
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
+    - uses: actions/checkout@v4
+    - uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
       with:
-        python-version: ${{ matrix.python-version }}
-    - uses: actions/checkout@v3
-    - name: Install node
-      uses: actions/setup-node@v3
-      with:
-       node-version: '18.x'
+        python_version: ${{ matrix.python-version }}
+        node_version: '18.x'
     - name: Install dependencies
-      run: python -m pip install -U jupyterlab~=3.0
-    - name: Build the extension
-      run: |
-        jlpm
-        jlpm run eslint:check
-        python -m pip install yt
-        python -m pip install .
-
+      run: python -m pip install -U jupyterlab~=4.0
+    - name: jlpm config
+      run: jlpm config set -H enableImmutableInstalls false
```

Comment:
Can't work out why, but `enableImmutableInstalls` was set to true, which causes an error when there is no yarn.lock file:

```
YN0028: The lockfile would have been created by this install, which is explicitly forbidden.
```

but the yarn file **should** be created here. so setting it to false seems correct. just not sure what caused the change in the CI test enviornment in this PR.

Reply:
We may want to commit our `yarn.lock` file.  I'm not sure if it's a universally-accepted practice for jupyterlab extensions ... but maybe I'm just out of the loop or woefully ignorant?!

Reply:
ya, i'm not sure either -- maybe we should be committing it? but I can research that separate from this PR.